pkg/ottl/ottlfuncs/README.md

```diff
@@ -338,6 +338,9 @@ The `replace_all_patterns` function replaces any segments in a string value or k
 
 If one or more sections of `target` match `regex` they will get replaced with `replacement`.
 
+The `replacement` string can refer to matched groups using [regexp.Expand syntax](https://pkg.go.dev/regexp#Regexp.Expand).
+Note that `$` may need to be escaped to avoid getting consumed by config file environment variable replacement.
```

Comment:
Could you document how to use match groups and insert a literal `$` given the environment variable expansion used in the Collector's config?